server/src/github.com/alexeyknyshev/cpsrv/cpsrv_patch_test.go

```diff
@@ -0,0 +1,425 @@
+	FreeAccess     *bool    `json:"free_access,omitempty"`
+	MainOffice     *bool    `json:"main_office,omitempty"`
+	WithoutWeekend *bool    `json:"without_weekend,omitempty"`
+	RoundTheClock  *bool    `json:"round_the_clock,omitempty"`
+	WorksAsShop    *bool    `json:"works_as_shop,omitempty"`
+	Schedule       Schedule `json:"schedule,omitempty"`
+	Tel            string   `json:"tel,omitempty"`
+	Additional     string   `json:"additional,omitempty"`
+	Rub            *bool    `json:"rub,omitempty"`
+	Usd            *bool    `json:"usd,omitempty"`
+	Eur            *bool    `json:"eur,omitempty"`
+	CashIn         *bool    `json:"cash_in,omitempty"`
+}
+
+type TestPatchReq struct {
+	Data   PatchRequest `json:"data"`
+	UserId uint         `json:"user_id"`
+}
+
+func getPatchExampleNewCP() *PatchRequest {
+	False := false
+	True := true
+	patchReq := PatchRequest{
+		Longitude:      37.6878262,
+		Latitude:       55.6946643,
+		Type:           "atm",
+		BankId:         322,
+		TownId:         4,
+		Address:        "г. Москва, Район Моей Мечты",
+		AddressComment: "ОАО UnderButtom",
+		MetroName:      "",
+		FreeAccess:     &True,
+		MainOffice:     &False,
+		WithoutWeekend: &False,
+		RoundTheClock:  &False,
+		WorksAsShop:    &True,
+		Schedule:       Schedule{},
+		Tel:            "",
+		Additional:     "",
+		Rub:            &True,
+		Usd:            &False,
+		Eur:            &False,
+		CashIn:         &False,
+	}
+	return &patchReq
+
+}
+
+func getPatchExampleExistCP() (*PatchRequest, string) {
+	patchReq := PatchRequest{
+		Id:     58552,
+		BankId: 2764,
+	}
+	exampleJson := "{\"schedule\":{},\"bank_id\":" + strconv.FormatUint(uint64(patchReq.BankId), 10) + "}"
+	return &patchReq, exampleJson
+
+}
+
+func searchLastPatch(t *testing.T, resJsonPatches []byte) uint32 {
+	var CPPatches map[string]interface{}
+	err := json.Unmarshal(resJsonPatches, &CPPatches)
+	if err != nil {
+		t.Errorf("Unmarshal err %v", err)
+	}
+	last_key := uint64(0)
+	//Search last patch number
+	for key := range CPPatches {
+		int_key, _ := strconv.ParseUint(key, 10, 64)
+		if int_key > last_key {
+			last_key = int_key
+		}
+	}
+	return uint32(last_key)
+}
+
+func comparePatches(t *testing.T, resPatch []interface{}, expectedPatch []interface{}) {
+	if len(resPatch) != len(expectedPatch) {
+		t.Error("response and expected patches have different field amount")
+		return
+	}
+
+	fields := []string{"patch id", "cashpoint id", "user_id", "data", "timestamp"}
+	for i, vol := range expectedPatch {
+		if i == 3 { //PATCH_DATA
+			checkJsonResponse(t, []byte(vol.(string)), []byte(resPatch[i].(string)))
+		} else if i != 4 { //don't check timestamp
+			if resPatch[i].(uint64) != vol.(uint64) {
+				t.Error("comparePatches: fields", fields[i], "don't match")
+			}
+		}
+	}
+}
+
+func invokeTaranPatchFuncs(t *testing.T, hCtx *HandlerContextStruct, requestJson []byte) (uint32, uint64) {
+	TaranResp, err := hCtx.Tnt().Call("cashpointProposePatch", []interface{}{requestJson})
```

Comment:
First part of with function is already implemented as [handlerCashpointCreate](https://github.com/alexeyknyshev/android_cash_points/blob/master/server/src/github.com/alexeyknyshev/cpsrv/handlers_cashpoints.go#L255)
It does not mean that your solution is wrong, but wrapping handler allow us to test http handler as well as tarantool-side logic.